packages/manager/src/features/CloudPulse/Alerts/AlertsListing/AlertListing.tsx

```diff
@@ -1,41 +1,75 @@
-import { TableRowLoading } from 'src/components/TableRowLoading/TableRowLoading';
-import { TableSortCell } from 'src/components/TableSortCell';
+import * as React from 'react';
+import { useHistory, useRouteMatch } from 'react-router-dom';
+
+import { DebouncedSearchTextField } from 'src/components/DebouncedSearchTextField';
 import { StyledPlaceholder } from 'src/features/StackScripts/StackScriptBase/StackScriptBase.styles';
 import { useAllAlertDefinitionsQuery } from 'src/queries/cloudpulse/alerts';
+import { useCloudPulseServiceTypes } from 'src/queries/cloudpulse/services';
 
-import { AlertTableRow } from './AlertTableRow';
-import { AlertListingTableLabelMap } from './constants';
+import { alertStatusOptions } from '../constants';
+import { AlertsListTable } from './AlertsListTable';
 
-import type { Alert } from '@linode/api-v4';
+import type { Item } from '../constants';
+import type { Alert, AlertServiceType, AlertStatusType } from '@linode/api-v4';
 
 export const AlertListing = () => {
-  // These are dummy order value and handleOrder methods, will replace them in the next PR
-  const order = 'asc';
-  const handleOrderChange = () => {
-    return 'asc';
-  };
-  const { data: alerts, isError, isLoading } = useAllAlertDefinitionsQuery();
-
+  const { url } = useRouteMatch();
   const history = useHistory();
+  const { data: alerts, error, isLoading } = useAllAlertDefinitionsQuery();
+  const {
+    data: serviceOptions,
+    error: serviceTypesError,
+    isLoading: serviceTypesLoading,
+  } = useCloudPulseServiceTypes(true);
 
-  const handleDetails = ({ id: _id, service_type: serviceType }: Alert) => {
-    history.push(
-      `${location.pathname}/detail/${serviceType}/${
-        _id
-      }`
-    );
-  };
+  const getServicesList = React.useMemo((): Item<
+    string,
+    AlertServiceType
+  >[] => {
+    return serviceOptions && serviceOptions.data.length > 0
+      ? serviceOptions.data.map((service) => ({
+          label: service.label,
+          value: service.service_type as AlertServiceType,
+        }))
+      : [];
+  }, [serviceOptions]);
+
+  React.useEffect(() => {
+    setServiceFilters(getServicesList);
+  }, [getServicesList]);
+  const [searchText, setSearchText] = React.useState<string>('');
+
+  const [serviceFilters, setServiceFilters] = React.useState<
+    Item<string, AlertServiceType>[]
+  >([]);
+  const [statusFilters, setStatusFilters] = React.useState<
+    Item<string, AlertStatusType>[]
+  >(alertStatusOptions);
+
+  const serviceFilteredAlerts = React.useMemo(() => {
+    if (serviceFilters && serviceFilters.length !== 0 && alerts) {
+      return alerts.filter((alert: Alert) => {
+        return serviceFilters.some(
+          (serviceFilter) => serviceFilter.value === alert.service_type
+        );
+      });
+    }
+
+    return alerts;
+  }, [serviceFilters, alerts]);
+
+  const statusFilteredAlerts = React.useMemo(() => {
+    if (statusFilters && statusFilters?.length !== 0 && alerts) {
```

Comment:
need to remove the redundant check here..
```suggestion
    if (statusFilters && statusFilters.length !== 0 && alerts) {
```